apps/internal/oauth/oauth.go

```diff
@@ -194,14 +195,31 @@ func (d DeviceCode) Token(ctx context.Context) (accesstokens.TokenResponse, erro
 	}
 }
 
-var waitRE = regexp.MustCompile("(authorization_pending|slow_down)")
+type deviceCodeError struct {
+	Error string `json:"error"`
+}
 
-// TODO(msal): This is freaking terrible. The original just looked for the exact word in the error output.
-// I doubt this worked. I don't know if the service really does this, but it should send back a structured
-// error response. Anyways, I updated this to search the entire return error message, which will be the body
-// of the return.
 func isWaitDeviceCodeErr(err error) bool {
-	return waitRE.MatchString(err.Error())
+	var c errors.CallErr
+	if errors.As(err, &c) {
+		if c.Resp.StatusCode != 400 {
+			return false
+		}
+		var dCErr deviceCodeError
+		defer c.Resp.Body.Close()
+		body, err := ioutil.ReadAll(c.Resp.Body)
+		if err != nil {
+			return false
+		}
+		err = json.Unmarshal(body, &dCErr)
+		if err != nil {
+			return false
+		}
+		if dCErr.Error == "authorization_pending" || dCErr.Error == "slow_down" {
```

Comment:
Has `slow_down` been removed?  I can't find it doc'ed https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code

Reply:
(we do this in our track 1 ADAL lib, I have no problem with having this check, more of a curiosity why it isn't documented)

Reply:
I think I see it in the official spec : https://tools.ietf.org/html/rfc8628#section-3.5 , there probably needs an update to the microsoft docs after verifiying it has not been removed